src/main/java/net/datafaker/transformations/JsonTransformer.java

```diff
@@ -117,19 +117,19 @@ private String generate(IN input, Collection<Object> collection) {
         return sb.toString();
     }
 
-    private static void value2String(Object value, StringBuilder sb) {
+    private void value2String(IN input, Object value, StringBuilder sb) {
         if (value == null) {
             sb.append("null");
         } else if (value instanceof Integer
-            || value instanceof Long
-            || value instanceof Short
-            || value instanceof BigInteger
-            || value instanceof Boolean
-            || (value instanceof Double
-            && BigDecimal.valueOf((Double) value).remainder(BigDecimal.ONE).doubleValue() == 0)
-            || (value instanceof BigDecimal
-            && ((BigDecimal) value).remainder(BigDecimal.ONE).doubleValue() == 0)) {
-            sb.append(value);
+                || value instanceof Long
+                || value instanceof Short
+                || value instanceof BigInteger
+                || value instanceof Boolean
+                || value instanceof Double
+                || value instanceof BigDecimal) {
+            // Quote if input type is specifically String
+            @SuppressWarnings("unused")
+            var tmp = input instanceof String ? sb.append('"').append(value).append('"') : sb.append(value);
```

Comment:
I'm not sure I like this, it was to address JsonTest where String is explicitly asked for. Though I'm not even sure we should facilitate that as it simply makes non-complaint (IMHO) JSON.
Ex (schema and transformer are specifically String):
```java
Schema<String, String> schema, String expected) {
        JsonTransformer<String> transformer = JsonTransformer.<String>builder().withCommaBetweenObjects(false).build();
```

Reply:
Right but it is for the JsonTest tests... the pre-existing ones

Reply:
It's definitely possible that I missed what was causing it 🤷‍♂️

Reply:
Well you can roll back the change and see the failure again 🙂

Reply:
I've changed this ternary to just `sb.append(value)`, and all tests are still green.

Reply:
Hmmm okay let me revisit it.... maybe maven cache or something